### PR TITLE
Feature/form

### PR DESCRIPTION
**Pull request by TheLuisAbarca**
1. Update and Fix misspells in column names of DB.
2. Add resources for posts, comments and likes on routes.
3. Implementation and updates of controllers in this project.
4. Update and Implement views for the 4 basic models that we have previously defined.
5. Add helpers and spec skeleton files for further use.
6. Fix rubocop linters.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -7,6 +7,25 @@ def index
...
+
+  def create
+    @user = User.find(params[:user_id])
+
+    @post = @user.posts.new(post_params)
+    if @post.save
+      redirect_to user_path(@user), notice: 'Successfully added a Post!'
+    else
+      flash.now[:notice] = 'Failed to create a Post.'
+      render :new
```
Comment (Dipeshtwis) [inline app/controllers/posts_controller.rb:26]: - [ ] I would recommend adding a `new` method to create an instance.

```ruby
class PostsController < ApplicationController

    ...

    def new
      @post = Post.new
    end
end
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    # @user = User.find(params[:author_id])
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new(comment_params)
+    @comment.author_id = @user.id
+
+    if @comment.save
+      redirect_to user_post_path(@user, @post), notice: 'Successful!'
+    else
+      flash.now[:notice] = 'Comment could not be created'
```
Comment (Dipeshtwis) [inline app/controllers/comments_controller.rb]: - [ ] Also here. I would recommend adding a `new` method to create an instance.

```ruby
class CommentsController < ApplicationController

    ...

    def new
      @comment = Comment.new
    end
end
```

File: config/routes.rb [hunk 3]
```diff
@@ -9,7 +11,12 @@
   # get 'users/:user_id/posts/:id', to: 'posts#show'
 
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :create]
+  end
+
+  resources :posts do
+    resources :comments, only: [:create]
+    resources :likes, only: [:create]
```
Comment (Dipeshtwis) [inline config/routes.rb]: - [ ] Also, add the route `new` for posts and comments.


```ruby
resources :posts, only: [:new, :create] do
  resources :comments, only: [:new, :create]
  ..
end
```

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class CommentsController < ApplicationController
+  before_action :find_post
+
+  def create
+    # @user = User.find(params[:author_id])
```
Comment (Dipeshtwis) [inline app/controllers/comments_controller.rb]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @TheLuisAbarca, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
:+1: The application is working fine. I'm able to create posts, comments, and likes.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (TheLuisAbarca) [reply app/controllers/posts_controller.rb:26]: Thanks, @Dipeshtwis for your kind suggestion and feedback!. This suggestion is implemented on commit 54ec112.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (TheLuisAbarca) [reply app/controllers/comments_controller.rb]: Thanks, @Dipeshtwis for your kind suggestion and feedback!. This suggestion is implemented on commit 54ec112.

File: config/routes.rb (hunk 3 above)
Comment (TheLuisAbarca) [reply config/routes.rb]: Thanks, @Dipeshtwis for your kind suggestion and feedback!. This suggestion is implemented on commit 54ec112.

File: app/controllers/comments_controller.rb (hunk 4 above)
Comment (TheLuisAbarca) [reply app/controllers/comments_controller.rb]: Thanks, @Dipeshtwis for your kind suggestion and feedback!. This suggestion is implemented on commit 54ec112.

**Comment by TheLuisAbarca** [comment]
Thanks, @Dipeshtwis  for taking your time and reviewing my job. :+1: :1st_place_medal:

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -26,4 +26,7 @@
       <%= render 'layouts/custom/comment_component', comment: comment %>
     <% end %>
   </div>
```
Comment (andersonlebon) [inline app/views/posts/show.html.erb:28]: ## [OPTIONAL]

- [ ] Kindly note that your project contains some errors that need to be fixed.  The route to show the post is not working.

image

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -6,7 +6,7 @@
     <p><%= @user.bio %></p>
   </div>
 
-  <% @posts.each do |post| %>
+  <% @user.most_recent_posts.each do |post| %>
     <%= render 'layouts/custom/post_component', post: post %>
   <% end %>
   <!--button><%= link_to 'See all posts', user_posts_path(@user) %></#button-->
```
Comment (andersonlebon) [inline app/views/users/show.html.erb:12]: ## [OPTIONAL]
- [ ] Kindly fix this error also when the See all post button is clicked. 
image

File: config/routes.rb [hunk 7]
```diff
@@ -9,7 +11,12 @@
   # get 'users/:user_id/posts/:id', to: 'posts#show'
 
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create]
+  end
+
+  resources :posts, only: [:new, :create] do
+    resources :comments, only: [:new, :create]
+    resources :likes, only: [:create]
```
Comment (andersonlebon) [inline config/routes.rb:19]: ## [OPTIONAL]

- [ ] Kindly try to structure very well your Routes.

**Review by andersonlebon — APPROVED** [review]
## Hi @TheLuisAbarca 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by TheLuisAbarca** [comment]
Thank you @andersonlebon for your kindly feedback.

**Pull request merged**